### PR TITLE
[FLINK-23438] Bump httpclient from 4.5.3 and 4.5.9 to 4.5.13

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Flink is still using org.apache.httpcomponents.httpclient:4.5.3 and org.apache.httpcomponents.httpclient:4.5.9. Those versions are impacted by CVE-2020-13956 (though Flink is not impacted by it). The latest available version is 4.5.13

## Brief change log

  - Updated httpclient dependencies from 4.5.3 and 4.5.9 to 4.5.13
  - Updated NOTICE files to deal with new version number
  
## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): don't know
  - Anything that affects deployment or recovery: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-filesystems/flink-s3-fs-hadoop/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -30,7 +30,7 @@ This project bundles the following dependencies under the Apache Software Licens
 - org.apache.hadoop:hadoop-common:3.1.0
 - org.apache.htrace:htrace-core4:4.1.0-incubating
 - org.apache.httpcomponents:httpcore:4.4.6
```
Comment (zentol) [inline flink-filesystems/flink-s3-fs-hadoop/src/main/resources/META-INF/NOTICE:32]: why are we not bumping httpcore as well?

File: flink-filesystems/flink-s3-fs-hadoop/pom.xml [hunk 2]
```diff
@@ -46,7 +46,7 @@ under the License.
 			<dependency>
 				<groupId>org.apache.httpcomponents</groupId>
 				<artifactId>httpclient</artifactId>
-				<version>4.5.9</version>
+				<version>4.5.13</version>
```
Comment (zentol) [inline flink-filesystems/flink-s3-fs-hadoop/pom.xml:49]: same comment as for flink-s3-fs-presto

File: flink-end-to-end-tests/flink-glue-schema-registry-test/pom.xml [hunk 3]
```diff
@@ -33,7 +33,7 @@ under the License.
 	<packaging>jar</packaging>
 
 	<properties>
-		<httpclient.version>4.5.9</httpclient.version>
+		<httpclient.version>4.5.13</httpclient.version>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-glue-schema-registry-test/pom.xml:36]: Are we not able to remove this property and the usages in this file?

File: flink-connectors/flink-connector-kinesis/pom.xml [hunk 4]
```diff
@@ -258,7 +258,7 @@ under the License.
 			<dependency>
 				<groupId>org.apache.httpcomponents</groupId>
 				<artifactId>httpclient</artifactId>
-				<version>4.5.9</version>
+				<version>4.5.13</version>
```
Comment (zentol) [inline flink-connectors/flink-connector-kinesis/pom.xml:261]: same comment as for the filesystems

File: flink-filesystems/flink-s3-fs-hadoop/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (MartijnVisser) [reply flink-filesystems/flink-s3-fs-hadoop/src/main/resources/META-INF/NOTICE:32]: I didn't want to create a PR with multiple dependencies. If you prefer, I can definitely include it. Let me know.

**Comment by MartijnVisser** [comment]
@zentol I'm testing the other review comments now, I'll get back to you when tests have completed

File: flink-filesystems/flink-s3-fs-hadoop/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (zentol) [reply flink-filesystems/flink-s3-fs-hadoop/src/main/resources/META-INF/NOTICE:32]: In practice it is part of httpclient so I would prefer to bump them at once.

File: flink-connectors/flink-connector-kinesis/pom.xml (hunk 4 above)
Comment (zentol) [reply flink-connectors/flink-connector-kinesis/pom.xml:248]: Why is this entry still there? Was there a problem after removing it?

File: flink-connectors/flink-connector-kinesis/pom.xml [hunk 5]
```diff
@@ -234,17 +232,6 @@ under the License.
 			<version>${aws.sdkv2.version}</version>
 		</dependency>
 
-		<dependency>
-			<groupId>org.apache.httpcomponents</groupId>
-			<artifactId>httpclient</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-connector-kinesis/pom.xml:239]: These should actually stay. This ensures that if a _user_ bundles this this connector in the jar they still get this version of the dependency. In that scenario what we define in the dependency management is completely ignored.

**Review by zentol — CHANGES_REQUESTED** [review]

**Review by zentol — APPROVED** [review]

**Pull request merged**